docs/_blog/bdk_rn_making_of.md

```diff
@@ -0,0 +1,238 @@
+
+```kotlin
+// File: BdkRnPackage.kt
+
+import com.facebook.react.ReactPackage
+import com.facebook.react.bridge.NativeModule
+import com.facebook.react.bridge.ReactApplicationContext
+import com.facebook.react.uimanager.ViewManager
+
+class BdkRnPackage : ReactPackage {
+
+    override fun createNativeModules(reactContext: ReactApplicationContext):
+            MutableList<NativeModule> {
+        return mutableListOf(BdkRnModule(reactContext))
+    }
+}
+
+```
+
+Now lets add code for creating a wallet in BdkRnModule.kt
+
+The methods used here are for bdk-kotlin and available in the bdk-kotlin documentation.
+
+We first create a key info object
+
+```kotlin
+// File: BdkRnModule.kt
+
+@ReactMethod
+fun createWallet(result: Promise) {
+  // Create key info with a new mnemonic
+  val keys: ExtendedKeyInfo = generateExtendedKey(
+        Network.TESTNET,
+        WordCount.WORDS12,
+        ""
+  )
+  
+  // more code to follow...
+  // create descriptor and change descriptor
+  // create databaseConfig and blockchainconfig
+  // create wallet
+  
+}
+```
+
+Then key info used to create a wallet descriptor and change descriptor:
+
+```kotlin
+val descriptor: String = "wpkh(" + keys.xprv + "/84'/1'/0'/0/*)"
+
+val changeDescriptor: String = descriptor.replace("/84'/1'/0'/0/*","/84'/1'/0'/1/*")
+```
+
+To create a wallet with bdk we need to specify wallet descriptor, network, a database config, blockchain config. We intend to use bitcoin testnet and want to use default memory for data. For bitcoin node we will use a public electrum server. We will need to define these parameters to create a wallet.
+
+```kotlin
+val network = `Network.TESTNET`
+val databaseConfig = DatabaseConfig.Memory
+blockchainConfig =
+      BlockchainConfig.Electrum(
+          ElectrumConfig("ssl://electrum.blockstream.info:60002", null, 5u, null, 10u)
+      )
+```
+
+Once done we can use these parameters to create a bdk wallet using the native android bdk library:
+
+```kotlin
+var wallet: BdkWallet = BdkWallet(
+          descriptor,
+          changeDescriptor,
+          setNetwork(network),
+          databaseConfig,
+          config
+)
+```
+
+Once we have a wallet initialised, we can call methods on it to sync, generate a new address and to get balance 
+
+```kotlin
+wallet.sync(ProgressLog, maxAddress)
+
+wallet.getNewAddress()
+
+wallet.getBalance().toLong()
+```
+
+To return a value from the native android code to React Native’s Javascript side over the JS Native bridge we will use `com.facebook.react.bridge.Promise`. To return balance information to JS, the following code can be used
+
+```kotlin
+val balance: String = wallet.getBalance().toLong()
+result.resolve(balance)
+```
+
+At this point  we have an Android native module and it can be invoked from JS by calling createWallet. if the wallet is created correctly it will return the balance.
```

Comment:
It think you should write createWallet here as `createWallet`. Also start the sentence `if ....` with caps.

Reply:
Done ✅